tracer/test/snapshots/SqlServerDatabaseMonitoringTests.enabled.verified.txt

```diff
@@ -0,0 +1,3530 @@
+﻿[
+  {
+    TraceId: Id_1,
+    SpanId: Id_2,
+    Name: sql-server.query,
+    Resource: DELETE FROM [System-Data-SqlClient-Test-GUID] WHERE Id=@Id;,
+    Service: Samples.SqlServer-sql-server,
+    Type: sql,
+    ParentId: Id_3,
+    Tags: {
+      component: SqlClient,
+      db.type: sql-server,
+      env: testing,
+      out.host: sqlsever,
+      runtime-id: Guid_1,
+      span.kind: client,
+      _dd.dbm_trace_injected: true
```

Comment:
Am I right in thinking this is the only change in the snapshots? Do we need to add a test somewhere to ensure we're correctly injecting the value, or are there already tests for that?

Reply:
Currently we don't have any other span metadata to test the injection because we don't want to show the comment in the span resource. We could introduce a test flag that runs the comment injection before span creation, that way we could see a difference in the span metadata, what do you think?

Reply:
Or we could have a test flag that adds another tag that shows the injected comment?

Reply:
I think adding an optional extra tag is probably the safest option if we go this route? 🤔 Messing with ordering sounds a bit more complex. 

If we add the extra tag I think we should probably use the "slow path", e.g. `span.AddTag("dbm_injected_comment", comment)` instead of adding a new property to `SqlTags` as the latter would increase allocations in all cases, where the former only degrades in the test scenario? `DD_INTERNAL_TRACE_TAG_DBM_COMMENT` or something like that to enable it?

That said, I don't know that we necessarily _have_ to have the comment shown in these snapshots, as long as we have unit tests for it elsewhere, but it does seem like a good way to end-to-end the injection across all the libraries and versions 🙂

Reply:
I like the idea above since it will allow us to verify that the data is being set as expected for the spans, but I also thought it was worth mentioning that the main goal of these snapshot test was to confirm that we are not breaking the customer's query when we do the propagation and the spans are returned as expected.

When it comes to adding this tag I am thinking it could make sense that I work on adding a test to the system-tests too/instead, there we can check that the comment is added to the query as expected using the DB/Agent container logs, the main thing I think about when it comes to the `DD_INTERNAL_TRACE_TAG_DBM_COMMENT` option is that we are not able to see the value that gets added to the query.

WDYT? Better if we add the comment to the tracer we have more control over this repo? 
Or does it make sense to you if instead the system-tests do the whole job of end to end testing?